### ISSUE TITLE
no-unused-new: should not be triggered

### ISSUE THREAD

**Issue by eamodio** [palantir/tslint#1839]
### Bug Report

- __TSLint version__: `4.0.2`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__:  VSCode

#### TypeScript code being linted

```ts
class Foo {
    value: string;
 }
class Bar {
    foo: Foo;
 }

function setFoo(bar: Bar) {
    bar.foo || (bar.foo = new Foo()); // lint error here (no-unused-new)
    bar.foo.value = 'hi';
}

let bar = new Bar();
setFoo(bar);
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "arrow-parens": false,
        "class-name": true,
        "comment-format": [
            false
        ],
        "curly": false,
        "indent": [
            true,
            "spaces"
        ],
        "new-parens": true,
        "no-duplicate-variable": true,
        "no-eval": true,
        "no-for-in-array": false,
        "no-internal-module": true,
        "no-reference": true,
        "no-trailing-whitespace": true,
        "no-unsafe-finally": true,
        "no-unused-expression": false,
        "no-unused-new": true,
        "no-var-keyword": true,
        "no-var-requires": true,
        "object-literal-key-quotes": [
            true,
            "as-needed"
        ],
        "one-line": [
            true,
            "check-open-brace",
            "check-whitespace"
        ],
        "one-variable-per-declaration": [
            true,
            "ignore-for-loop"
        ],
        "ordered-imports": [
            false,
            {
                "named-imports-order": "case-insensitive"
            }
        ],
        "quotemark": [
            true,
            "single",
            "avoid-escape"
        ],
        "radix": true,
        "semicolon": [
            true,
            "always"
        ],
        "trailing-comma": [
            true,
            {
                "multiline": "never",
                "singleline": "never"
            }
        ],
        "triple-equals": [
            true,
            "allow-null-check"
        ],
        "typedef-whitespace": [
            true,
            {
                "call-signature": "nospace",
                "index-signature": "nospace",
                "parameter": "nospace",
                "property-declaration": "nospace",
                "variable-declaration": "nospace"
            }
        ],
        "use-isnan": true,
        "variable-name": [
            true,
            "allow-leading-underscore",
            "allow-pascal-case",
            "ban-keywords",
            "check-format"
        ],
        "whitespace": [
            true,
            "check-branch",
            "check-decl",
            "check-operator",
            "check-module",
            "check-separator",
            "check-type"
        ]
    }
}
```

#### Actual behavior

[tslint] do not use 'new' for side effects (no-unused-new)

#### Expected behavior

No lint error

**Comment by andy-hanson** [comment]
The implementation of `no-unused-new` seems overly complicated. It looks inside of an expression for a `new` subexpression. This apparently catches `5 + new Foo()`, but that's not really useful. The type checker will catch that and so will `no-unused-expression`. Why not deprecate `no-unused-new` and make it an option of `no-unused-expression`?
